src/Provider/Github.php

```diff
@@ -62,6 +62,29 @@ public function getResourceOwnerDetailsUrl(AccessToken $token)
         return $this->domain . '/api/v3/user';
     }
 
+    protected function fetchResourceOwnerDetails(AccessToken $token)
+    {
+        $response = parent::fetchResourceOwnerDetails($token);
+
+        if (($response['email'] ?? null) == null) {
+            $url = $this->getResourceOwnerDetailsUrl($token) . '/emails';
+
+            $request = $this->getAuthenticatedRequest(self::METHOD_GET, $url, $token);
+
+            $responseEmail = $this->getParsedResponse($request);
+
+            if (false === is_array($responseEmail)) {
+                throw new \UnexpectedValueException(
+                    'Invalid response received from Authorization Server. Expected JSON.'
+                );
+            }
+
+            $response['email'] = $responseEmail[0]['email'];
```

Comment:
```suggestion
            $response['email'] = $responseEmail[0]['email'] ?? null;
```

I don't think there is a need for the type check at all. If the request fails then there will already be an error.